tests/upgrades/test_activation_key.py

```diff
@@ -0,0 +1,143 @@
+"""Test Activation Key related Upgrade Scenario's
+
+:Requirement: Upgraded Satellite
+
+:CaseAutomation: Automated
+
+:CaseLevel: Acceptance
+
+:CaseComponent: API
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+
+from nailgun import entities
+from requests.exceptions import HTTPError
+from robottelo.test import APITestCase
+from upgrade_tests import pre_upgrade, post_upgrade
+
+
+class scenario_positive_activation_key(APITestCase):
+    """Activation key is intact post upgrade and verify activation key update/delete
+
+    :steps:
+
+        1. In Preupgrade Satellite, create activation key with different entities
+        2. Upgrade the satellite to next/latest version
+        3. Update existing activation key with different entities
+        4. Postupgrade, Verify the activation key is intact, update and delete
+
+    :expectedresults: Activation key should update and delete successfully.
```

Comment:
under 'expectedresults' please include pre-upgrade result too. i.e. Activation key should create .